### PR TITLE
bring back dedicated log in button to make log in more usable

### PR DESCRIPTION
**Pull request by jancborchardt**
We had a lot of people saying they were confused with the arrow button and it also came up in a few tests I did. So here’s the dedicated log in button back.

Before & after:
capture du 2016-10-06 16-48-47capture du 2016-10-06 16-47-13
(Deliberately in french so we can check if it’s understandable without words. ;)

Please review @nextcloud/designers @LukasReschke @oparoz @nextcloud/accessibility

### CODE REVIEW

**Comment by tcitworld** [comment]
Not sure the *stay connected* checkbox should be over the *Connect* button. I'm more used to the opposite.

**Comment by phsc** [comment]
I'd prefer @jancborchardt suggestion.

The reason is, that I use the tab key to navigate from "username" to "password". And with the next click on tab, I directly get to the "stay connected" checkbox, which I can activate with the space key. And finally I press enter to login.
If you would change the order, I had to press tab twice to get from "password" to "stay connected". I know, I'm lazy... ;-)

**Comment by tcitworld** [comment]
> If you would change the order, I had to press tab twice to get from "password" to "stay connected". I know, I'm lazy... ;-)

I think you can choose the tab order of the elements separately from their display.

**Comment by oparoz** [comment]
The one thing I'm not sure about is the design of the button itself. I find it a bit too discreet and not looking enough like a button.

**Comment by Espina2** [comment]
I suggest that you switch the actual UI language to what I did in the homepage and more similar to other issue I open. (But this is me) 

Its time to start to do the same things = .

I think is far better you have the button, in mobile for example it helps a lot. :)

**Comment by jancborchardt** [comment]
Good points everyone! Modified it to this:
capture du 2016-10-06 19-09-09

- Round style of button makes it seem more like a button. Still needs to be adjusted in the rest of Nextcloud of course ;)
- Tab index order still goes with the »Stay logged in« first. Not sure if that’s better because it’s different than the DOM order
- In any case, checking the »stay logged in« will automatically trigger log in when the details are filled out. ;)

**Comment by MariusBluem** [comment]
Can we do this optional 😪 ... or can we make a poll in the Forum?

**Comment by jancborchardt** [comment]
> Can we do this optional 😪 ... or can we make a poll in the Forum? 

We don’t do polls on the forum about technical decisions either, this is not how a project works. ;)
Thing is that as said above this has come up as an issue many times. A dedicated button is what people are used to and what is more obvious. Experimenting with the way it currently is was nice but it resulted in a hindered UX. Introducing a setting for this would be a bit ridiculous. ;)

File: core/css/styles.css [hunk 1]
```diff
@@ -616,8 +594,9 @@ label.infield {
 	width: 249px;
 }
 #body-login input.login {
-	width: auto;
-	float: right;
+	width: 269px;
+	background-position: right 16px center;
+	border-radius: 50px;
```
Comment (Espina2) [inline core/css/styles.css]: In general don't give width to the buttons. 

You can copy this code for having the same style and hover. 

You have this if you use scss

https://gist.github.com/Espina2/65b1099e603cf517096e41a5eae07578

or you can use this compiled version

https://gist.github.com/Espina2/3f803075c3306076298076d807f43157

**Comment by Espina2** [comment]
@jancborchardt I think the arrow have to be more center. 

In general don't give width to the buttons let the margins do the work. 

You can copy this code for having the same style and hover. 

You have this if you use scss

https://gist.github.com/Espina2/65b1099e603cf517096e41a5eae07578

or you can use this compiled version

https://gist.github.com/Espina2/3f803075c3306076298076d807f43157

**Comment by jancborchardt** [comment]
@Espina2 can I ask you to add a commit which adds that behavior? :) It seems that it’s enough to change width to margin, and the background from having it on the element to the `:before` element.

(Note that we use CSS without preprocessor in all of Nextcloud so it’s easily accessible.)

File: core/css/styles.css (hunk 1 above)
Comment (tcitworld) [inline core/css/styles.css]: Oh, well, seems logical. Still find the radius - on the website and here - a little too big.

**Comment by Espina2** [comment]
> @Espina2 can I ask you to add a commit which adds that behavior? :) It seems that it’s enough to change width to margin, and the background from having it on the element to the :before element.
> 
> (Note that we use CSS without preprocessor in all of Nextcloud so it’s easily accessible.)

I dont have this project locally.

**Comment by juliushaertl** [comment]
I don't like the round button in combination with the current Nextcloud design. They look quite bulky compared to the rest of the UI. What about using the first button with small rounded corners and making it a bit smaller:

2016-10-07-093150_955x579_scrot

**Comment by MorrisJobke** [comment]
@jancborchardt the signed-off comment is missing in your commit: https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work (that causes CI to fail)

**Comment by jancborchardt** [comment]
@MorrisJobke so if I use `git commit -ms "comment bla"` it’ll work?

@juliushaertl yeah, I tend to agree that the circular-end buttons don’t look nice with the current design. It’s a bit too much design for its own sake and feels a bit dated.
What we did before is have the »Stay logged in« and Log in button next to each other, button on the right. But that doesn’t work with all languages. I’ll make it have automatic width and be floated to the right, as confirmation buttons usually are.

Like, since we’re quoting other platforms for log in designs, basically none of them use rounded-off buttons. cc @Espina2 what do you think?

**Comment by MorrisJobke** [comment]
> @MorrisJobke so if I use git commit -ms "comment bla" it’ll work?

Correct, but I  guess you need to do `git commit -sm "comment bla"` because `-m` requires the comment as very next parameter ;)

**Comment by Espina2** [comment]
> @juliushaertl yeah, I tend to agree that the circular-end buttons don’t look nice with the current design. It’s a bit too much design for its own sake and feels a bit dated.
> What we did before is have the »Stay logged in« and Log in button next to each other, button on the right. But that doesn’t work with all languages. I’ll make it have automatic width and be floated to the right, as confirmation buttons usually are.

You have a circular logo so you are a circular brand, you should use circular rounded things when its possible. 

The problem itself its not the button, its you dont have a clean are to put them. If all the people like the image why not having something more pleasant for the eyes like this. (You can ignore the sign in for now) 

normal

@juliushaertl if haved take a look in the issue/post that I post after this issue are open you can see that I pull some examples. Somehow my post is gone or closed, I cant find out to show you if you didnt see it.

**Comment by jancborchardt** [comment]
@Espina2 again, as that is basically the same screen you showed before in https://github.com/nextcloud/server/issues/1615#issuecomment-251367185: sure the overall look is like on the website, but the streamlining is missing. It’s too many tiny elements with way too much information, even with the »Stay logged in« checkbox missing. Things are separated and the visual hierarchy is confusing. The labels like »Log in« and »In order to continue …« are not necessary. And lastly the big white container around everything simply looks a bit outdated.

It seems like you might think that the current design has no design reasoning behind it and you just want to throw it away completely. That’s not really how things work out well. We developed this over time through iterations and tests and know what worked well and what doesn’t (for example, the removal of the log in button didn’t work so well).
Doing a _complete_ throwaway just on a mockup basis is not a very good idea design-wise for several reasons: If you change everything, you don’t know what improved, and there’s a lot which can also not work out. Also you change the complete flow of a software which millions of people are using in one step. That’s bound to cause confusion.

Hope that doesn’t come off as aggressive or such because it’s not meant to be. We do have to work together here though and simply changing everything just because is not how it works in a big project. There needs to be proper reasoning.


In the line of thought of gradual improvement, how about this? It also addresses the point of the missing »Forgot password?« link that you brought up and @pierreozoux mentioned.
capture du 2016-10-07 13-10-20


Anyway – I’d like more of @nextcloud/designers to chime in and also @karlitschek @LukasReschke @MorrisJobke @ChristophWurst etc. Maybe I’m missing something here, but currently this diverges a lot from the simplicity of the existing Nextcloud design.

**Comment by jancborchardt** [comment]
Alternatively with a rounded log in button:
capture du 2016-10-07 13-14-50
A »Create new account« could be added to the footer. We can also move »Create new account« and »Forgot your password?« to the footer to keep focus on the important elements.

**Comment by juliushaertl** [comment]
Just as @jancborchardt said, we should't just drop everything for something completely new. There should be a step-by-step iteration with a good reason behind each change, like adding the login button.

> You have a circular logo so you are a circular brand, you should use circular rounded things when its possible.

This only applies as long as the Nextcloud logo is in place. 
On the new Nextcloud website those rounded/circle centered design looks really nice @Espina2 since every element has been redesigned to fit, but I don't think that this style fits well for a software user interface.

Another point on theming: if users start to customize their Nextcloud installation, the logo is a very common change i think. So in my opinion we should keep the user interface as neutral as possible.

**Comment by Espina2** [comment]
So since I don't know how I'm doing I only complete want to change the Nextcloud app without explain nothing. Lets start

1 - Research 

Box
screen shot 2016-10-07 at 12 51 49

One Drive
screen shot 2016-10-07 at 12 52 55

Google Drive
screen shot 2016-10-07 at 12 53 27

Quiver
screen shot 2016-10-07 at 12 54 07


What they are doing? Simple and clean login page without any images and visual noise very easy to find where to do things. All they have button and work well in mobile. 

What I do it in my version? Remove the noisy background, put all the things visible and turn the process of create an account easier. Like the others softwares related do. I think I dont need to explain more on this. 

> It seems like you might think that the current design has no design reasoning behind it and you just want to throw it away completely. That’s not really how things work out well. We developed this over time through iterations and tests and know what worked well and what doesn’t (for example, the removal of the log in button didn’t work so well).
> Doing a complete throwaway just on a mockup basis is not a very good idea design-wise for several reasons: If you change everything, you don’t know what improved, and there’s a lot which can also not work out. Also you change the complete flow of a software which millions of people are using in one step. That’s bound to cause confusion.

For you knowing what works best you need to change it. If you don't changed you never know. We are not changing nothing conceptual we are creating cohesion, you still have the same elements in the page you only add two new ones (for making the things easier). 

> This only applies as long as the Nextcloud logo is in place. 
On the new Nextcloud website those rounded/circle centered design looks really nice @Espina2 since every element has been redesigned to fit, but I don't think that this style fits well for a software user interface.
 
@juliushaertl You are wrong, and I can show many projects that you want. So when you are excluding the Nextcloud logo you are personalize at your own taste, the same thing should be to the whole product (like the buttons, fonts etc...) its impossible to have one solution that will work for all the Styles and personalizations that the people are doing. 

So I don´t like to write long text's to explain all the things that I do. 

It good to say that I am not a junior designer, I am a senior designer with many projects out there, and I win one or two awards. I can't be so wrong in all the things that I am doing in the past years... 

So to clarify what I am saying I will help if my help is needed if you think that everything is fine right now you can  just ignore and I at sometime I will stop posting "new things". 

I am very straight forward person and I say what I think. And @jancborchardt I'm glad that you do the same. I don't feel bad that it sounds aggressive I know its difficult to change the work of another person, that I made a lot of effort on it, but embrace it, you are having new ideias for free. :)

In the end who the have last word is you not me.

**Comment by phsc** [comment]
Why not combine both proposals?

Take @jancborchardt latest proposal with the rounded button (very clean and simple) and place it in a whitebox as in @Espina2's proposal (more pleasant for the eyes).
This way, it was a good improvement without a complete throwaway of the current design

**Comment by skjnldsv** [comment]
I must say I disagree with the round button too. Overall, the nextcloud design is very "retro". I'm not here to change everything and nextcloud works fine as it is. Nevertheless, having a great and modern design show that the software isn't derelict. When I see the buttons, the public pages, the login, i say to myself "we should really redesign this" (to be honest I also say this to a lot of websites 😝)

But this is a common thing that we encounter among opensource projects. The overall design is neglected (strong word here, I should exaggerate speaking of nextcloud, but I will leave it for my example) in profit of the good behavior/function of the app. I have a lot of example in mind of apps I use a lot that are terrible UI speaking. :/

Anyhow, I really think nextcloud is an awesome project and would love to see more designs like @Espina2's (new website, news, article and login...) There's something here I really like!

**Comment by oparoz** [comment]
I love having all these screenshots. It's only when you can compare that you realise some things are not quite right.
So far I like @Espina2 's login page best. It immediately focuses your attention on what needs to get done and should offer the right compromise for both, people you want and don't want a background image.
Looking at the current login page (and the one for OneDrive), you get the feeling that the CSS is broken and that the background for the login form is missing.
But it's all subjective of course.

**Comment by oparoz** [comment]
I just wanted to add that I also agree with @jancborchardt . The new design has to fit with the rest, so maybe we should first re-introduce the button and then try and make the whole thing look better and especially **bring back contrast** :)

**Comment by skjnldsv** [comment]
Yeah we should follow the overall design. But this rounded button don't fit. The first one is closer to what we're used to. :)

**Comment by Espina2** [comment]
I have show the problems and bring some solutions and examples that I think it could be. You can make whatever you want with information I put here. So now I just open the spotify and I see that they for example use rounded buttons in their app UI. 

And just for curiosity I see that they do it the exact same way in the website. See print screen's above. 

And look why they used it. Maybe because the logo is a circle? Everything is a circle. 

screen shot 2016-10-07 at 15 44 01
screen shot 2016-10-07 at 15 43 09

**Comment by MorrisJobke** [comment]
The font weight of the button text is too big. I would go for the normal 300 instead of 600.

**Comment by skjnldsv** [comment]
Much better @jancborchardt !

**Comment by raghunayyar** [comment]
Apologies for a late reply but here are my two cents on this debate.

This is really a huge thread so I couldn't complete reading the debate. I am in favour of having a separate button but completely against have rounded buttons, simply because we don't use them anywhere else and hence are not a part of the brand identity / design guidelines. A retro / modern feel is relative. Spotify uses a lot of buttons on its website as rounded and hence has really less number of rectangular ones so it also has a design guide, just that it is not similar to us.

The examples of different websites stated is not a right way because they are truly open source and a certain marketing mind goes behind designing them in order to sell it. Mailchimp for example has a massive image on the login page whereas Spotify has just whitespace on their login page. So, these are not great examples to refer either because you can find an example for every requirement.

If it is a "look nice" reasoning or the logo being circle logic behind having circular buttons, I don't think its a good approach. A lot of new UI components can be made cooler, but that is not how it works as it leads to less focus on the design language. I would like the buttons to be consistent across the project, in their raw form at least.

Design is subjective, its not a good idea to fight over rounded or squared buttons. We have a lot of design issues and a lot of areas which need design love, we should focus on those.

**Comment by Espina2** [comment]
@raghunayyar 

> This is really a huge thread so I couldn't complete reading the debate. I am in favour of having a separate button but completely against have rounded buttons, simply because we don't use them anywhere else and hence are not a part of the brand identity / design guidelines. A retro / modern feel is relative. Spotify uses a lot of buttons on its website as rounded and hence has really less number of rectangular ones so it also has a design guide, just that it is not similar to us.

We are trying to create design guidelines so it will change the rectangular for the circular. Right now we don't have nothing of this. And retro modern is relative I dont think so. The thing or look retro or look modern you can have the two because are the opposite. If can you show one example where you have the two things I'm glad to see.

And for now the buttons are used on the website, and they are rounded because it makes sense. You are a rounded brand with circles and on the otherside there is no reason for using rectangular ones only the personal taste of who made it.

> The examples of different websites stated is not a right way because they are truly open source and a certain marketing mind goes behind designing them in order to sell it. Mailchimp for example has a massive image on the login page whereas Spotify has just whitespace on their login page. So, these are not great examples to refer either because you can find an example for every requirement.

I will glad to see better examples, can you show them?

For the example of mailchimp, they only use image when you are on desktop's.
screen shot 2016-10-15 at 16 04 05

> 
If it is a "look nice" reasoning or the logo being circle logic behind having circular buttons, I don't think its a good approach. A lot of new UI components can be made cooler, but that is not how it works as it leads to less focus on the common design language. I would like the buttons to be consistent across the project, in their raw form at least.
 
This is why we are thinking why the would be, for be consistent and all the apps, websites. For you design anything you have to think why you use that style not only because you like or I like.

> Design is subjective, its not a good idea to fight over rounded or squared buttons. We have a lot of design issues and a lot of areas which need design love, we should focus on those.

Design is objective and subjective at the same time, because you doing the things for a purpose, you are not making art... 

Of course you more issues, but we are trying to resolve one here.

**Comment by raghunayyar** [comment]
Sign in for mailchimp:
screen shot 2016-10-15 at 6 24 00 pm

**Comment by raghunayyar** [comment]
> Design is subjective, its not a good idea to fight over rounded or squared buttons. We have a lot of design issues and a lot of areas which need design love, we should focus on those.

I personally genuinely don't think round and square button debate is an issue which deserves a lot of time, there are are several usability issues / enhancements Nextcloud needs which might increase more users using the product than a replacing a square button with a rounded one. 

Design is not objective. Round logo means round buttons is not actually a rule of thumb impregnated somewhere not following which would we fail at design. Design evolves subjectively  and to write these guidelines we present the current state of art and not do it from scratch by having entirely new components and yes, a rounded button is a new component.

Talking about buttons,  what if I start arguing why we use #ffffff for backgrounds and not #f8f8f8, or why we use #333333 for content and not #222222 we won't move ahead with this and I can tell as a designer there is no objective reasoning to either of these. These are subjective decisions made earlier and its better to stick to such decisions in order to move ahead in an open environment.

**Comment by enoch85** [comment]
What a shame that we have to bring back the login button and remove the clean experience that we had before IMHO. With that being said, I really like @Espina2 login screen.

**Comment by Espina2** [comment]
> Design is not objective. Round logo means round buttons is not actually a rule of thumb impregnated somewhere not following which would we fail at design. Design evolves subjectively and to write these guidelines we present the current state of art and not do it from scratch by having entirely new components and yes, a rounded button is a new component.

You are mixing two things togheter design is subjective and objective like I said. If you analyze data and design with that, it are very objective and things are made because you know that this are a better way to do it.

At the moment design is far better objective with the evolution of user experience and research. So I can have your opinion and its okay. Just pointing some facts. 

And for the fight if it would be rounded or square or rectangle buttons I only presente a valid point to why I think they should be rounded? Your only reason its to point that in the they are rectangle but you dont give me a reason for them being rectangle. Just saying that you are not thinking in what they should be, you are point only that they are this...


> I personally genuinely don't think round and square button debate is an issue which deserves a lot of time, there are are several usability issues / enhancements Nextcloud needs which might increase more users using the product than a replacing a square button with a rounded one.

> Talking about buttons, what if I start arguing why we use #ffffff for backgrounds and not #f8f8f8, or why we use #333333 for content and not #222222 we won't move ahead with this and I can tell as a designer there is no objective reasoning to either of these. These are subjective decisions made earlier and its better to stick to such decisions in order to move ahead in an open environment.

This doesnt start about buttons, this start in that point https://github.com/nextcloud/server/issues/1615

You should have reason to use that colors, for less or more contrast for exemple. I really believe and I make things thinking why I make them. 

And for the image on the mailchimp its very objective that you first look the image and after that only you see the login. So if you test I will say that without the image most of the people make login more faster. And if you think this are subjective you can test it...

So I don't like for doing anything I have to have like a political or philosophy debate.

**Comment by jancborchardt** [comment]
Folks, this discussion is getting too long and fruitless. One of our main tenets is to be pragmatic, and to do things in small steps. As said above we will simply not throw away our current log in screen, lots of people are used to it and themed it etc. The image is nearly all blue so not very distracting.

If we compare with other web services it makes little sense to just compare with the log in page. We need to compare with their home page, which you usually go to when you want to sign in. And for example [Slack](https://slack.com/) which everyone loves so much has a big fat nice image on there.

I updated the PR and postponed the »Forgot password?« link for later. @MorrisJobke all our buttons have semibold font because they are calls to action. With 300 font-weight they look strange, and also don’t fit the icons.

Please review @skjnldsv @raghunayyar @oparoz @pierreozoux @MorrisJobke @juliushaertl – Let’s get this in so we can move forward.

**Comment by skjnldsv** [comment]
All good to me!
Thanks @jancborchardt :)

👍

**Comment by Espina2** [comment]
@jancborchardt 

Just to people hate me more the slack login page don't have image. Not even in the "app" they have. The homepage its not the real login page. :)

And sorry for be such an annoying. 

image
image

**Comment by jancborchardt** [comment]
@Espina2 I know. That’s why I said:

> If we compare with other web services it makes little sense to just compare with the log in page. We need to compare with their home page, which you usually go to when you want to sign in.

**Comment by Espina2** [comment]
I know. :)

**Comment by juliushaertl** [comment]
:+1: 

Let's get this in. :smile:

**Comment by jancborchardt** [comment]
Thanks everyone. ;) At least I guess we can all agree that the arrow icon is much better now ;)

**Comment by enoch85** [comment]
Going backwards design wise IMHO. But not my call...

**Comment by jancborchardt** [comment]
@enoch85 design is about making the product easy to use for people. Sure the previous log in screen looked »cleaner« for people like us who use Nextcloud every day. However we are also very likely to just stay logged in and rarely ever see the screen.

People who don’t use it that often have problems discovering it. And if they already have problems at such a basic thing as the log in screen, we failed. We had multiple people make remarks and open issues about it, and I also watched people use Nextcloud which confirmed the problem.

So this is definitely a step forward. And as always – if there’s disagreement or downvotes, please state your reasoning. Just saying »I don’t like it« or »it’s a step backwards« has no basis, and is likely wrong. :wink:  It’s always recommended to watch »normal« people use a software to get a feel for how understandable it is. For that also read [»Watch people use the software«](http://jancborchardt.net/usability-in-free-software#Watch-people-use-the-software). :smiley:

**Comment by enoch85** [comment]
@jancborchardt Yeah, I totally get the reason, just don't like it :)

**Comment by eppfel** [comment]
@enoch85 You could still make it optional in the Theming App or make your own App 😉

**Pull request merged**